File: web/src/types/packagingRelation.ts
/* 包装关系模块类型（对齐 OpenAPI 契约） */
export interface PackagingRelation {
  id: number;
  parentLevelId: number;
  parentLevelName?: string | null;
  childLevelId: number;
  childLevelName?: string | null;
  childQty: number;
  isFixedQty: 0 | 1;
  tareWeight?: number | null;
  status: 0 | 1;
  createdAt: string;
  updatedAt: string;
  createdBy?: number | null;
  updatedBy?: number | null;
}

export interface PackagingRelationSummary {
  id: number;
  parentLevelId: number;
  parentLevelName?: string | null;
  childLevelId: number;
  childLevelName?: string | null;
  childQty: number;
  isFixedQty: 0 | 1;
  status: 0 | 1;
  createdAt: string;
}

export interface PackagingRelationListResponse {
  items: PackagingRelationSummary[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

export interface CreatePackagingRelationRequest {
  parentLevelId: number;
  childLevelId: number;
  childQty: number;
  isFixedQty?: 0 | 1;
  tareWeight?: number;
  status?: 0 | 1;
}

export interface UpdatePackagingRelationRequest {
  childQty?: number;
  isFixedQty?: 0 | 1;
  tareWeight?: number;
}

export interface PackagingRelationQueryParams {
  parentLevelId?: number;
  status?: 0 | 1;
  page?: number;
  pageSize?: number;
}

File: api/src/main/java/com/aistore/module/packagingrelation/entity/PackagingRelation.java
package com.aistore.module.packagingrelation.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/** 包装关系实体（表 packaging_relation） */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("packaging_relation")
public class PackagingRelation {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;
    private Long parentLevelId;
    private Long childLevelId;
    private BigDecimal childQty;
    private Integer isFixedQty;
    private BigDecimal tareWeight;
    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    private Long createdBy;
    private Long updatedBy;

    @TableLogic
    private Integer deleted;
}
